cf/test/test_external.py

```diff
@@ -161,6 +161,41 @@ def test_EXTERNAL_WRITE(self):
         for i in range(len(h)):
             self.assertTrue(external[i].equals(h[i], verbose=2))
 
+    def test_EXTERNAL_AGGREGATE(self):
+        if self.test_only and inspect.stack()[0][3] not in self.test_only:
+            return
+
+        # Read parent file without the external file, taking first field to test
+        f = cf.read(self.parent_file, verbose=0)[0]
+        measure_name = 'measure:area'
+
+        # Split f into parts (take longitude with 3 x 3 = 9 points) so can
+        # test the difference between the aggregated result and original f.
+        # Note all parts retain the external variable cell measure.
+        f_lon_thirds = [f[:, :3], f[:, 3:6], f[:, 6:]]
+
+        g = cf.aggregate(f_lon_thirds)
+
+        self.assertEqual(len(g), 1)
+
+        # Check cell measure construct from external variable has been removed
+        self.assertFalse(g[0].cell_measures())
+
+        # Check aggregated field is identical to original with measure removed
+        f0 = f.copy()
+        f0.del_construct(measure_name)
+        self.assertEqual(g[0], f0)
+
+        # Now try aggregating when one part doesn't have the cell measure; we
+        # expect the two with the measure to aggregate but the other not to
```

Comment:
@davidhassell I thought you might want to review this PR to check its implementing the behaviour discussed. If not, or you don't have time etc., please just merge.

In preparing this PR I did realise something that didn't come up in our discussion the other day that might warrant a slight change of approach, or if not at the least will be something to consider when we come back to consider external cell measures in future.

In short, the fact we are deleting the external cell measure on the aggregated fields post-aggregation means that the output fields from `cf.aggregate` could, in these cases involving external cell measure, potentially be aggregatable further still (i.e. if they were input to `cf.aggregate` once more they could be reduced to a smaller FieldList) because the removal changes them and the reduced nature means they might become aggregatable with fields they previously were not with. 

I've commented against this part of code, specifically the basic test, I added because it forms an example: the output `g` has two fields according to our agreed approach, but if those fields were input to `cf.aggregate` again they would aggregate down to one.

I believe the whole point of aggregate is to combine fields as much as possible according to the aggregation rules in a single operation. So perhaps due to that we need to add in a further logical step to account for external cell measures whereby, in the case an external cell measure is present on an aggregated field and has to be deleted, the fields that would otherwise be returned all have to be evaluated once more for aggregation? Or it might be that I can move/change the deletion logic meaning we don't end up with this problem in the first place. What would you advise?

Reply:
Hi Sadie, this is an interesting point, and I think you're right. I guess the answer would be to remove external measures _before_ the signature is created, then, rather than just noting that they exist and dealing with them later. That would involve a field copy, of course, but we could do the copy inside the `for key, msr in f.cell_measures.items():` loop, so only those affected fields are copied. 

How does that sound? Perhaps it's worth updating the PR now for this?

Reply:
Good idea, I think that sounds best with all options considered. I'll get the PR updated accordingly. Thanks!